iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ReplaceTeePointByVoltageLevelOnLine.java

```diff
@@ -179,48 +182,9 @@ public void apply(Network network, boolean throwException,
 
-        TopologyKind topologyKind = tappedVoltageLevel.getTopologyKind();
-        if (topologyKind == TopologyKind.BUS_BREAKER) {
-            Bus bus = tappedVoltageLevel.getBusBreakerView().getBus(bbsOrBusId);
-            if (bus == null) {
-                notFoundBusInVoltageLevelReport(reporter, bbsOrBusId, tappedVoltageLevel.getId());
-                LOGGER.error("Bus {} is not found in voltage level {}", bbsOrBusId, tappedVoltageLevel.getId());
-                if (throwException) {
-                    throw new PowsyblException(String.format("Bus %s is not found in voltage level %s", bbsOrBusId, tappedVoltageLevel.getId()));
-                } else {
-                    return;
-                }
-            }
-            Bus bus1 = tappedVoltageLevel.getBusBreakerView()
-                    .newBus()
-                    .setId(newLine1Id + "_BUS_1")
-                    .add();
-            Bus bus2 = tappedVoltageLevel.getBusBreakerView()
-                    .newBus()
-                    .setId(newLine2Id + "_BUS_2")
-                    .add();
-            createBusBreakerSwitches(bus1.getId(), bus.getId(), bus2.getId(), bbsOrBusId, tappedVoltageLevel.getBusBreakerView());
-            newLine1Adder.setBus2(bus1.getId());
-            newLine1Adder.setConnectableBus2(bus1.getId());
-            newLine2Adder.setBus1(bus2.getId());
-            newLine2Adder.setConnectableBus1(bus2.getId());
-        } else if (topologyKind == TopologyKind.NODE_BREAKER) {
-            BusbarSection bbs = tappedVoltageLevel.getNodeBreakerView().getBusbarSection(bbsOrBusId);
-            if (bbs == null) {
-                notFoundBusbarSectionInVoltageLevelReport(reporter, bbsOrBusId, tappedVoltageLevel.getId());
-                LOGGER.error("Busbar section {} is not found in voltage level {}", bbsOrBusId, tappedVoltageLevel.getId());
-                if (throwException) {
-                    throw new PowsyblException(String.format("Busbar section %s is not found in voltage level %s", bbsOrBusId, tappedVoltageLevel.getId()));
-                } else {
-                    return;
-                }
-            }
-            int bbsNode = bbs.getTerminal().getNodeBreakerView().getNode();
-            int firstAvailableNode = tappedVoltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;
-            createNodeBreakerSwitches(firstAvailableNode, firstAvailableNode + 1, bbsNode, "_1", newLine1Id, tappedVoltageLevel.getNodeBreakerView());
-            createNodeBreakerSwitches(firstAvailableNode + 3, firstAvailableNode + 2, bbsNode, "_2", newLine2Id, tappedVoltageLevel.getNodeBreakerView());
-            newLine1Adder.setNode2(firstAvailableNode);
-            newLine2Adder.setNode1(firstAvailableNode + 3);
+        // Create the breaker topology
+        if (!createTopology(newLine1Adder, newLine2Adder, tappedVoltageLevel, reporter, throwException)) {
```

Comment:
I just realized that this code is a big duplication of ConnectVoltageLevelOnLine :scream:

We should keep in mind to factorize this later.

Reply:
yep good point, we might want to call the other network modification or something close